optimizely/event/event_processor.py

```diff
@@ -59,14 +61,14 @@ class BatchEventProcessor(BaseEventProcessor):
 
     def __init__(
         self,
-        event_dispatcher,
-        logger=None,
-        start_on_init=False,
-        event_queue=None,
-        batch_size=None,
-        flush_interval=None,
-        timeout_interval=None,
-        notification_center=None,
+        event_dispatcher: Optional[type[EventDispatcher] | CustomEventDispatcher] = None,
+        logger: Any = None,
```

Comment:
Can we use TypeVar instead of any type because data type will be lost once it is stored in any typed variable. 
```suggestion
        Logger = TypeVar("Logger")
        logger: Logger = None,
```
or if we already know which types to allow then
```suggestion
        Logger = TypeVar("Logger", int, string)
        logger: Logger = None,
```